docs/pages.mjs

```diff
@@ -1,47 +1,65 @@
-import path, { dirname } from "path";
-import { fileURLToPath } from "url";
-import fs from "fs";
-
-function getFiles(dir) {
-  return fs.readdirSync(dir).flatMap((item) => {
-    const currentPath = path.join(dir, item);
-    if (fs.statSync(currentPath).isDirectory()) {
-      return {
-        dir: true,
-        name: item,
-        path: currentPath,
-      };
-    }
+import { readdir, readFile, stat } from 'node:fs/promises'
+import { join } from 'node:path'
+import { fileURLToPath } from 'node:url'
 
-    return {
-      dir: false,
+/**
+ * Scans the directory for files and collects them
+ * @param {import("node:fs").PathLike} dir The directory to scan the files from
+ * @returns {Promise<Array<{ dir: boolean; name: string; path: string; }>>} The files in the directory
+ */
+async function getFiles(dir) {
+  /**
+   * @type Array<{ dir: boolean; name: string; path: string; }>
+   */
+  const result = []
+  const dirAsPathString = dir instanceof URL ? fileURLToPath(dir) : dir
+
+  const dirContents = await readdir(dir)
+
+  for (const item of dirContents) {
+    const currentPath = join(dirAsPathString, item)
+
+    const currentPathStat = await stat(currentPath)
+
+    result.push({
```

Comment:
Why take the mutable approach and push to an array instead of keeping the immutable `flatmap`?

Reply:
Primarily because using `flatMap` means you're returning Promises inside Promises. This is because `map` and `flatMap` use callback functions which themselves have to be async, regardless of whether the function they are used in is already async or not. This is just a PITA you really don't want to have to deal with when doing Promises.

Other than that, it is fairly common knowledge that on average the order of fastest looping functions is:
1. `for...i`
2. `for...of`
3. `map` / `flatMap`

While speed is of little concern with the pages this particular script is looping to, other users of magidoc might have many more pages and then it does become something to consider.

Reply:
Okay I see. I don't think the speed difference much a concern if there is one, but I'm okay with keeping it like that. Thanks for the explanation.